src/repr/src/adt/interval.rs

```diff
@@ -127,23 +134,27 @@ impl Interval {
             return None;
         }
 
-        let seconds =
-            self.dur_as_secs() as f64 / other + months.fract() * 60.0 * 60.0 * 24.0 * 30.0;
-        if seconds.is_nan() || seconds < i64::MIN as f64 || seconds > i64::MAX as f64 {
+        let days = self.days as f64 / other + months.fract() * 30.0;
+        if days.is_nan() || days < i32::MIN as f64 || days > i32::MAX as f64 {
             return None;
         }
 
-        let nanos = self.nanoseconds() as f64 / other + seconds.fract() * 1e9;
-        if nanos.is_nan() || nanos < i64::MIN as f64 || nanos > i64::MAX as f64 {
+        let micros = self.micros as f64 / other + days.fract() * 1e6 * 60.0 * 60.0 * 24.0;
+        if micros.is_nan() || micros < i64::MIN as f64 || days > i64::MAX as f64 {
             return None;
         }
 
-        Self::new(months as i32, seconds as i64, nanos as i128).ok()
+        Self::new(months as i32, days as i32, micros as i64).ok()
     }
 
     /// Returns the total number of whole seconds in the `Interval`'s duration.
```

Comment:
The comment here and below now seem misleading. If I were to read "total" I'd expect months and days to be apart of it. Can you reword?

Reply:
I agree, I ended up removing these functions.